File: filter_MATLAB/write_int.m
function write_int(file_name,data)
% -------------------------------------
% 保存小端存储的二进制float类型的干涉图数据
% 输入参数：
%   file_name   保存的干涉图数据文件名
%   data        干涉图复数矩阵
% -------------------------------------

fid=fopen(file_name,'w');
data=data';
data=data(:);
[m,n]=size(data);
imag_data=sin(data);
real_data=cos(data);
count=m*2*n;
data=zeros(count,1);
data(1:2:count)=real_data;
data(2:2:count)=imag_data;
fwrite(fid,data,'float32');
fclose(fid);

File: test_Python/write_int.py
import numpy as np

def write_int(file_name, data):
    data = data.T
    data = data.ravel()
    imag_data = np.sin(data)
    real_data = np.cos(data)
    count = 2 * len(data)
    output_data = np.zeros(count)
    output_data[0::2] = real_data
    output_data[1::2] = imag_data
    output_data.astype(np.float32).tofile(file_name)
